plugin/ignite-plugin/src/main/scala/org/finos/vuu/feature/ignite/schema/SchemaMapper.scala

```diff
@@ -32,33 +32,22 @@ object SchemaMapper {
   }
 
   private def hasUniqueColumnNames(columnNames: List[String]): ValidationError = {
-    if (columnNames.distinct.size != columnNames.size)
-      return Some(s"Fields map contains duplicated column names")
-
-    None
+    Option.when(columnNames.distinct.size != columnNames.size)(s"Fields map contains duplicated column names")
   }
 
   private def externalFieldsInMapConformsToExternalSchema(externalSchema: ExternalStoreEntitySchema,
                                                           externalFields: Iterable[String]): ValidationError = {
-    externalFields.foreach(field => {
-      val exists = externalSchema.schemaFields.exists(_.name == field)
-      if (!exists) return Some(s"Field `$field` not found in external schema")
-    })
-
-    None
+    externalFields
+      .find(field => externalSchema.schemaFields.forall(_.name != field))
+      .map(f => s"Field `$f` not found in external schema")
   }
 
   private def internalFieldsInMapMatchTableColumns(columns: Array[Column],
                                                    internalFields: Iterable[String]): ValidationError = {
-    internalFields.foreach(columnName =>
-      if (!columns.exists(_.name == columnName))
-        return Option(s"Column `$columnName` not found in internal columns")
-    )
-
-    if (columns.length > internalFields.size)
-      return Option(s"More internal columns passed than mapped fields")
-
-    None
+    internalFields
+      .find(columnName => columns.forall(_.name != columnName))
+      .map(columnName => s"Column `$columnName` not found in internal columns")
+      .orElse(Option.when(columns.length > internalFields.size)(s"More internal columns passed than mapped fields"))
```

Comment:
Not sure having no mapping for a column is an error case.
If we look at BasketConstituent example, there are several csv files, one for each basket. It could be valid that one of the basket is missing any values for a BasketConstituent field

Reply:
I was unaware of this edge case. Thanks for pointing this out.